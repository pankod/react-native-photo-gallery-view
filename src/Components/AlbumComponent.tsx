// Global Imports
import React, { Component } from 'react';
import { Text, View, Image, TouchableOpacity, Dimensions, ScrollView, FlatList } from 'react-native';

// Local Imports
import { AlbumStyle } from "@Styles";
import { IAlbumProps, IMediaItem, IGalleryProps, IGalleryState } from "@Interfaces";
import Common from '@Provider';

const { width, height } = Dimensions.get('window');

export class AlbumComponent extends Component<IAlbumProps, {}> {

	static contextType = Common

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<FlatList style={AlbumStyle.container} data={context.mediaList}
							numColumns={context.gridSize}
							renderItem={({ item, index }) => this.renderItem(item, index)}
							keyExtractor={(item, index) => index.toString()}
						/>
					)
				}
			</Common.Consumer>
		)
	}

	private renderItem(item: IMediaItem, index: number): JSX.Element {
		const { gridSize, onSelectionChanged } = this.context;
		return (
			<TouchableOpacity onPress={() => onSelectionChanged(item, index)} key={index} style={{ width: width / gridSize, height: width / gridSize }}>
				<Image resizeMode={"cover"} style={{ flex: 1 }} key={index} source={{ uri: item.thumb }} />
			</TouchableOpacity>
		)
	}
}
