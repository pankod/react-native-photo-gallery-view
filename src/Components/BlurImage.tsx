// Global Imports
import React, { PureComponent } from 'react'
import { Animated } from 'react-native'

// Local Imports
import { IBlurImageProps } from '@Interfaces';
import { BlurImageStyle } from '@Styles';

export class BlurImage extends PureComponent<IBlurImageProps> {

	public imageAnimated = new Animated.Value(0);

	private onImageLoad(): void {
		Animated.timing(this.imageAnimated, {
			toValue: 1,
			duration: 500,
			useNativeDriver: true
		}).start();
	}

	render() {
		return (
			<React.Fragment>
				<Animated.Image
					{...this.props}
					style={[BlurImageStyle.image, { opacity: this.imageAnimated }]}
					onLoad={this.onImageLoad.bind(this)}
					blurRadius={1}
				/>
			</React.Fragment>
		)
	}
}
