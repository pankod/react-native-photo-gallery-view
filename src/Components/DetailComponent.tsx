// Global Imports
import React, { PureComponent } from 'react'

// Local Imports
import Common from '@Provider';
import { IGalleryProps, IGalleryState } from '@Interfaces';
import { BlurImage } from '@Components';

export class DetailComponent extends PureComponent {

	static contextType = Common;

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<BlurImage
							resizeMethod={"resize"}
							resizeMode={"contain"}
							source={{ uri: context.showingImage && context.showingImage.photo }}
							thumbnail={{ uri: context.showingImage && context.showingImage.thumb }}
						/>
					)
				}
			</Common.Consumer>
		)
	}
}
