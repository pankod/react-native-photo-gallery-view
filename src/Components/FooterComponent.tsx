// Global Imports
import React, { PureComponent } from 'react';
import { View } from 'react-native';

// Local Imports
import { FooterStyle } from "@Styles";
import { IFooterProps, IGalleryProps, IGalleryState } from "@Interfaces";
import Common from '@Provider';
import { Const } from '@Constants';

export class FooterComponent extends PureComponent<IFooterProps, {}> {

	static contextType = Common;

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<View style={[FooterStyle.container, context.customFooterStyle]}>
							{context.renderStickyFooter && this.renderStickyFooter()}
						</View>
					)
				}
			</Common.Consumer>
		)
	}

	private renderStickyFooter(): JSX.Element {
		return (
			this.context.renderStickyFooter(Const.HEIGHT)
		)
	}
}
