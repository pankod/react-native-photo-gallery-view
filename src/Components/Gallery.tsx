import React, { Component } from "react"
import { Text, View } from "react-native"

export class Gallery extends Component {
	render() {
		return (
			<View>
				<Text> textInComponent </Text>
			</View>
		)
	}
}
