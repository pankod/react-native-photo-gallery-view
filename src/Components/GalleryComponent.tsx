// Global Imports
import React, { Component } from "react"
import { View } from "react-native"

// Local Imports
import { TopBarComponent, AlbumComponent, FooterComponent } from "@Components";
import { IGalleryProps, IGalleryState, IMediaItem } from "@Interfaces";
import { GalleryStyle } from "@Styles";
import { Const } from '@Constants';
import Common from '@Provider';

export class GalleryComponent extends Component<IGalleryProps, IGalleryState> {

	constructor(props: IGalleryProps) {
		super(props);
		this.state = {
			title: `${props.mediaList.length} Photos`
		}
	}

	static defaultProps = {
		stickyFooter: true,
		gridSize: 3,
	}

	public render(): JSX.Element {
		const {
			style,
			stickyFooter
		} = this.props;
		return (
			<Common.Provider value={{ ...this.state, ...this.props }}>
				<View style={[GalleryStyle.container, style]}>
					<TopBarComponent />
					<AlbumComponent />
					{stickyFooter &&
						<FooterComponent />
					}
				</View>
			</Common.Provider>
		)
	}
}
