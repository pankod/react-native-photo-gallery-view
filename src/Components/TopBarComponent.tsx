// Global Imports
import React, { Component } from 'react';
import { Text, View, TouchableWithoutFeedback, Image } from 'react-native';

// Local Imports
import { TopBarStyle } from '@Styles';
import { ITopBarProps, IGalleryState, IGalleryProps } from "@Interfaces";
import Common from '@Provider';

export class TopBarComponent extends Component<ITopBarProps, {}> {

	static contextType = Common;

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<View style={[TopBarStyle.container, context.customTopBarStyle]}>
							{this.backButtonRender()}
							{context.title && <Text>{context.title}</Text>}
						</View>
					)
				}
			</Common.Consumer>
		)
	}

	public backButtonRender(): JSX.Element {
		const { customTopBarBackIcon } = this.context;
		if (customTopBarBackIcon) {
			return (
				<TouchableWithoutFeedback style={TopBarStyle.backBtnStyle}>
					<Image source={customTopBarBackIcon} />
				</TouchableWithoutFeedback>
			)
		}

		return (
			<TouchableWithoutFeedback style={TopBarStyle.backBtnStyle} onPress={() => this.context.onBack && this.context.onBack()}>
				<Image source={require('../Assets/Images/back.png')} style={TopBarStyle.backBtnImageStyle} />
			</TouchableWithoutFeedback>
		)
	}
}
