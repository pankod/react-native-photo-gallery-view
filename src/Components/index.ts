export { GalleryComponent } from '@Components/GalleryComponent';
export { TopBarComponent } from '@Components/TopBarComponent';
export { FooterComponent } from '@Components/FooterComponent';
export { AlbumComponent } from '@Components/AlbumComponent';
export { BlurImage } from '@Components/BlurImage';
export { DetailComponent } from '@Components/DetailComponent';
export { PreviewModalComponent } from '@Components/PreviewModalComponent';
