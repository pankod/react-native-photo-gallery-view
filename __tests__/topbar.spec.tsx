
import * as React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';
import renderer from "react-test-renderer";
import { ShallowWrapper, shallow, mount } from "enzyme";

import { TopBarComponent } from "../src/Components/TopBarComponent";
import Common from '../src/Provider';

describe("Topbar Component", () => {
	let wrapper: ShallowWrapper;
	let mounting;
	const props = {
		onBackRequest: jest.fn(() => true)
	}
	const state = {
		selected: [],
	}
	let component = (
		<Common.Provider value={{ mediaList: [], gridSize: 3, ...props, ...state }}>
			<TopBarComponent {...props} />
		</Common.Provider>
	);

	beforeEach(() => {
		wrapper = shallow(component, { context: { ...props, ...state, gridSize: 3, mediaList: [] } });
		mounting = mount(component, { context: { ...props, ...state, gridSize: 3, mediaList: [] } });
	});

	test('should Topbar Component is render correctly', () => {
		const render = renderer.create(component).toJSON();

		expect(render).toBeDefined();
		expect(render).toMatchSnapshot();
	});

	test('should titleRender return customSelectedTitle or default title', () => {
		const spy = spyOn(TopBarComponent.prototype, "titleRender").and.callThrough();
		const instance = mounting.instance() as any;

		instance.context.selected = [0];
		expect(instance.titleRender().type.displayName).toEqual("Text");

		instance.context.customSelectedTitle = jest.fn(() => <View />);
		expect(instance.titleRender()).toBeTruthy();

		instance.context.selected = [];
		instance.context.isModalOpen = true;
		expect(instance.titleRender().type.displayName).toEqual("Text");

		instance.context.customDetailTitle = jest.fn(() => <View />);
		expect(instance.titleRender()).toBeTruthy();

		instance.context.customDetailTitle = undefined;
		instance.context.isModalOpen = false;
		expect(instance.titleRender().type.displayName).toEqual("Text");

		instance.context.customMainTitle = jest.fn(() => <View />);
		expect(instance.titleRender()).toBeTruthy();

		expect(spy).toHaveBeenCalled();
	});

	test('should fired backButtonRender return customTopBarBackButton or default button', () => {
		const spy = spyOn(TopBarComponent.prototype, "backButtonRender").and.callThrough();
		const instance = mounting.instance() as any;

		instance.backButtonRender();
		instance.context.customTopBarBackButton = jest.fn(() => <View />);
		expect(instance.backButtonRender()).toBeTruthy();

		instance.context.customTopBarBackButton = undefined;
		instance.context.onBackRequest = jest.fn(() => true);
		expect(instance.backButtonRender().type.displayName).toEqual("TouchableWithoutFeedback");

		expect(spy).toHaveBeenCalled();
	})


});
