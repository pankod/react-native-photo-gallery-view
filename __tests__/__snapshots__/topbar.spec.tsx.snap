// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar Component should Topbar Component is render correctly 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock />
  </react-native-mock>
</react-native-mock>
`;
